packages/labs/analyzer/src/lib/errors.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ts from 'typescript';
+
+const diagnosticsHost: ts.FormatDiagnosticsHost = {
+  getCanonicalFileName(name: string) {
+    return name;
+  },
+  getCurrentDirectory() {
+    return process.cwd();
+  },
+  getNewLine() {
+    return '\n';
+  },
+};
+
+export class DiagnosticsError extends Error {
+  diagnostics: ts.Diagnostic[];
+  constructor(diagnostics: readonly ts.Diagnostic[], message?: string);
```

Comment:
Do we need all these overloads? It doesn't seem like they are used.

Reply:
There's only two ways to call it (with an array of `diagnostics` or with a node and an error message)... most are the latter (where we're making a new one), but I added the former for this case, to format diagnostics coming from the initial compilation: https://github.com/lit/lit/pull/2976/files#diff-d0ce248fdc93fd9bbffed64c12cf21d16650a57ad613cbd739be0f03646c4059R86

I figured it would be nice to have them all as one error type with a `diagnostics` field so you could e.g. `try { ... } catch (e) { if (e instanceof DiagnosticError) { /* do something with `e.diagnostics` */ }`